master/internal/api_project.go

```diff
@@ -41,57 +45,216 @@ func (a *apiServer) GetProjectByID(
 func (a *apiServer) getProjectColumnsByID(
 	ctx context.Context, id int32, curUser model.User,
 ) (*apiv1.GetProjectColumnsResponse, error) {
-	notFoundErr := status.Errorf(codes.NotFound, "project (%d) not found", id)
-	p := &projectv1.Project{}
-	if err := a.m.db.QueryProto("get_project", p, id); errors.Is(err, db.ErrNotFound) {
-		return nil, notFoundErr
-	} else if err != nil {
-		return nil, errors.Wrapf(err, "error fetching project (%d) from database", id)
-	}
-	if ok, err := project.AuthZProvider.Get().CanGetProject(ctx, curUser, p); err != nil {
+	p, err := a.GetProjectByID(ctx, id, curUser)
+	if err != nil {
 		return nil, err
-	} else if !ok {
-		return nil, notFoundErr
-	}
-	// Get general columns
-	generalColumns := make([]projectv1.GeneralColumn, 0, len(projectv1.GeneralColumn_value))
-	for gc := range projectv1.GeneralColumn_value {
-		generalColumns = append(
-			generalColumns, projectv1.GeneralColumn(projectv1.GeneralColumn_value[gc]))
 	}
-	// Get hyperpatameters columns
-	hyperparameters := struct {
-		Hyperparameters []string
+
+	columns := []*projectv1.ProjectColumn{
+		{
+			Column:      "id",
+			DisplayName: "ID",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "name",
+			DisplayName: "Name",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "description",
+			DisplayName: "Description",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "tags",
+			DisplayName: "Tags",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT, // is this right?
```

Comment:
style: can either we remove this comment and `// ?` below, or elaborate on them so they make sense to the future readers of the code?

Reply:
will do!